### PR TITLE
New package: audit-2.8.4 (closes #2832)

### LINKED ISSUE #2832: Package Request: audit
[Linux Audit Security Framework](https://salsa.debian.org/debian/audit.git)

Preliminary template:

```
# Template file for 'audit'
pkgname=audit
version=2.8.4
revision=1
build_style=gnu-configure
configure_args="--sbindir=/usr/bin --libdir=/usr/lib --with-python --with-python3
 --enable-shared=audit --enable-gssapi-krb5 --with-apparmor --with-libcap-ng
 --with-prelude --with-libwrap"
hostmakedepends="automake libtool pkg-config intltool
 swig python-setuptools python3-setuptools"
makedepends="mit-krb5-devel libldap-devel libapparmor-devel libcap-ng-devel
 python-devel python3-devel libprelude-devel libwrap-devel"
short_desc="Linux Security Auditing Framework"
maintainer="Cameron Nemo <camerontnorman@gmail.com>"
license="GPL-3.0-or-later"
homepage="https://people.redhat.com/sgrubb/audit"
distfiles="${homepage}/${pkgname}-${version}.tar.gz"
checksum=badbadbadbadbadbadbadbadbadbadbadbadbadbadbadbadbadbadbadbadbadb
```

Remaining tasks:

- [ ] Package [libprelude](https://www.prelude-siem.org/)
- [ ] Package [TCP Wrapper / libwrap](ftp://ftp.porcupine.org/pub/security/index.html)
- [ ] Create subpackages for auditd, libauparse, libaudit, python{,3}-audit, audispd-plugins, golang bindings (see [debian packaging](https://salsa.debian.org/debian/audit/blob/debian/debian/control))

### PR DESCRIPTION
**Pull request by CameronNemo**

### CODE REVIEW

File: srcpkgs/audit/template [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        pkg_install() {
+                vmove "usr/lib/libaudit.so.*"
+        }
+}
+
+libaudit-devel_package() {
+        short_desc+=" - Library development files"
+        depends="libaudit-${version}_${revision}"
+        pkg_install() {
+                vmove usr/include/libaudit.h
```
Comment (maxice8) [inline srcpkgs/audit/template]: indent is wrong here

File: srcpkgs/audit/files/auditctl/run [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh -e
+
+test ! -r /etc/default/auditctl || . /etc/default/auditctl
+
+case "$USE_AUGENRULES" in
```
Comment (maxice8) [inline srcpkgs/audit/files/auditctl/run]: i suppose it would be nice if it could source 'conf' like it is done on wpa_supplicant to get this value

**Comment by Hoshpak** [comment]
I have finished a preliminary version of the musl patch: https://github.com/Hoshpak/void-packages/blob/CameronNemo-audit/srcpkgs/audit/patches/musl.patch

With this patch, audit compiles with musl but only if the python bindings are disabled. There seems to be an issue with swig under musl, the compiler fails to detect the function `SWIG_from_wchat_t` and thus fails to compile the bindings. I will also look into that but couldn't find anything yet.

**Comment by Hoshpak** [comment]
I updated the patch to fix the python bindings for musl.

File: srcpkgs/audit/template [hunk 3]
```diff
@@ -0,0 +1,116 @@
+# Template file for 'audit'
+pkgname=audit
+version=2.8.4
+revision=1
+build_style=gnu-configure
+configure_args="--sbindir=/usr/bin --libdir=/usr/lib --enable-shared=audit
```
Comment (Hoshpak) [inline srcpkgs/audit/template]: `--sbindir=/usr/bin` is already defined by the build_style, it can be omitted here.

**Pull request closed without merging**